### PR TITLE
Made IBM docker repo as default and provided an option to change to desired repo

### PR DESCRIPTION
**Pull request by jay-katta**
Made IBM docker repo as default and provided an option to change to desired repo

### CODE REVIEW

**Comment by PatrickDuncan** [comment]
Hey @jay-katta, with agent-slim and agent-kmodule there would be 2 places in the daemonset file that would need to change.

Easy way would be to modify your sed to change the repo path and keep the image name the same.

```shell
sed -i.bak -e "s|\( *image: \)icr.io/ext/sysdig/|\1$SYSDIG_REPO|g"
```

You add a final / after `$SYSDIG_REPO` if you want

**Comment by PatrickDuncan** [comment]
@jay-katta You would also need to copy over the `default-icr-io` secret into the $NAMESPACE and then add it to the daemonset file. 
```yaml
spec:
  ...
  template:
    spec:
      ...
      imagePullSecrets:
      - name: default-icr-io
```

File: agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh [hunk 1]
```diff
@@ -231,6 +233,11 @@ function install_k8s_agent {
         rm /tmp/sysdig-agent-daemonset-v2.yaml.bak
     fi
 
+    if [ ! -z "$SYSDIG_REPO" ]; then
+        sed -i.bak -e 's|\(.*image: icr.io/ext/sysdig/agent\)|        image: '$SYSDIG_REPO'|g' /tmp/sysdig-agent-daemonset-v2.yaml
```
Comment (mbreitung) [inline agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh]: if the indentation changes in the upstream yaml, this sed will break

Comment (mbreitung) [reply agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh]: the use of embedded single quotes is unnecessary. Please use double quotes around the expression and no quotes inside

Comment (mbreitung) [reply agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh]: I think that this sed can be tighter
`"s|\(.*\)image:\(.*\)|/1image: $SYSDIG_REPO|g"`

**Comment by PatrickDuncan** [comment]
Hey @mbreitung this should be closed because of #80

**Comment by mbreitung** [comment]
closing in favor of #80

**Pull request closed without merging**